clientv3/integration/kv_test.go

```diff
@@ -113,6 +114,39 @@ func TestKVPut(t *testing.T) {
 	}
 }
 
+// TestKVPutWithIgnoreValue ensures that Put with WithIgnoreValue does not clobber the old value.
+func TestKVPutWithIgnoreValue(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 1})
+	defer clus.Terminate(t)
+
+	kv := clientv3.NewKV(clus.RandClient())
+
+	_, err := kv.Put(context.TODO(), "foo", "", clientv3.WithIgnoreValue())
+	if grpc.ErrorDesc(err) != etcdserver.ErrKeyNotFound.Error() {
+		t.Fatalf("err expected %v, got %v", etcdserver.ErrKeyNotFound, err)
+	}
+
+	if _, err := kv.Put(context.TODO(), "foo", "bar"); err != nil {
```

Comment:
@heyitsanthony i feel this is required to ensure the functionality is working fine. The main reason for adding this feature was to support this very thing of detaching lease without passing the value and also support other features along with it. Just checking for key exists is not sufficient, since we are not sure whether its actually detaching the lease now. May be working now but later to ensure it doesn't break?